### PR TITLE
Extract `handle_missing_response!`

### PR DESCRIPTION
**Pull request by ecoologic**
* Fix build badge (by removing EOL ruby version)
* Extract method to keep consistent level of abstraction
* Add an extra debug log (that pairs with the one in `handle_response`)

Happy to rebase it or break it down in multiple prs if you like.

### CODE REVIEW

File: lib/devise_openid_authenticatable/strategy.rb [hunk 1]
```diff
@@ -55,6 +50,16 @@ def handle_response!
       end
     end
 
+    def handle_missing_response!
+      logger.debug("Delegate authentication to Rack::OpenID by throwing a 401")
+
+      opts = { identifier: identity_url, return_to: return_url, trust_root: trust_root, method: 'post' }
```
Comment (ecoologic) [inline lib/devise_openid_authenticatable/strategy.rb]: I have taken the liberty of converting this hash to 1.9 syntax

File: lib/devise_openid_authenticatable/strategy.rb [hunk 2]
```diff
@@ -55,6 +50,16 @@ def handle_response!
...
+      opts[:immediate] = true if scoped_params["immediate"]
+      opts[:optional]  = mapping.to.openid_optional_fields if mapping.to.respond_to?(:openid_optional_fields)
+      opts[:required]  = mapping.to.openid_required_fields if mapping.to.respond_to?(:openid_required_fields)
```
Comment (ecoologic) [inline lib/devise_openid_authenticatable/strategy.rb]: And here I aligned the equal signs

**Comment by nbudin** [comment]
Looks good, thanks for the PR!

**Comment by ecoologic** [comment]
The merge was really quick!, @nbudin thank you for the gem!

**Comment by nbudin** [comment]
Sure thing @ecoologic!  Do you want me to make a release with this change as well?  (I'm not sure what functionality it is enabling, it just seems like a nice code cleanup - but if you have projects depending on this change being released I'll be happy to cut a new version.)

**Comment by ecoologic** [comment]
Right now I'm just studying the gem to use it in a rails prj.  Maybe officially dropping 1.9 is a  CHANGELOG worthy note.

This PR is plain refactoring (except for the added debug log), it doesn't seem very urgent to me, so, this is not affecting me, I leave that to you.

**Comment by nbudin** [comment]
OK, cool, in that case I will leave the 1.9 drop for the next feature release.  Thanks again for the cleanup!

**Pull request merged**